File: client/src/api/product.ts
import api from './index'
import type { ApiResult } from '@/types'

export interface ProductItem {
  id: number
  title: string
  description: string
  price: number
  originalPrice: number | null
  condition: string
  images: string
  categoryId: number
  sellerId: number
  sellerName: string
  status: string
  auditStatus: string
  viewCount: number
  favCount: number
  createdAt: string
}

export interface ProductDetail extends ProductItem {
  categoryName: string
  sellerAvatar: string
}

export function getProducts(params?: { keyword?: string; categoryId?: number; page?: number; pageSize?: number }) {
  return api.get<ApiResult<ProductItem[]>>('/products', { params })
}

export function getProductById(id: number) {
  return api.get<ApiResult<ProductDetail>>(`/products/${id}`)
}

export function createProduct(data: {
  title: string
  description: string
  price: number
  originalPrice?: number
  condition: string
  images: string[]
  categoryId: number
}) {
  return api.post<ApiResult<ProductItem>>('/products', data)
}

export function updateProduct(id: number, data: Record<string, unknown>) {
  return api.put<ApiResult<ProductItem>>(`/products/${id}`, data)
}

export function deleteProduct(id: number) {
  return api.delete<ApiResult<null>>(`/products/${id}`)
}

export function getMyProducts() {
  return api.get<ApiResult<ProductItem[]>>('/products/my')
}

export function uploadImage(file: File) {
  const form = new FormData()
  form.append('file', file)
  return api.post<ApiResult<string>>('/upload/image', form, {
    headers: { 'Content-Type': 'multipart/form-data' },
  })
}

File: server/src/main/java/com/campustrade/entity/Product.java
package com.campustrade.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("product")
public class Product {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String title;
    private String description;
    private BigDecimal price;
    private BigDecimal originalPrice;
    @TableField("`condition`")
    private String condition;
    private String images; // JSON string
    private Long categoryId;
    private Long sellerId;
    private String status;
    private String auditStatus;
    private String auditReason;
    private Integer viewCount;
    private Integer favCount;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}
